### PR TITLE
12200 - do less attribute lookups on twisted.web.http

### LINKED ISSUE #12200: Remove attribute lookups from twisted.web
#12171 has performance issues, probably because of many attribute lookups on `twisted.web.http`, e.g. in `twisted.web.server`. We should remove those, and just do "from twisted.web.http import X".

### PR DESCRIPTION
**Pull request by itamarst**
## Scope and purpose

Fixes #12200 

The hope is this will make #12171 less of a performance regression.

### CODE REVIEW

**Comment by itamarst** [comment]
please review

File: src/twisted/web/server.py [hunk 1]
```diff
@@ -34,7 +34,14 @@
 from twisted.spread.pb import Copyable, ViewPoint
 from twisted.web import http, iweb, resource, util
```
Comment (adiroiban) [inline src/twisted/web/server.py:35]: Do we still want to keep the `http` import  here ? Just asking :)
 
I guess that this is here to simplify the access to the HTTP response code.

We can leave this PR as it is.

In the future, maybe we can have a separare "collection" object for the response codes, other than the `twisted.web.http` module itself.

```suggestion
from twisted.web import iweb, resource, util
```

**Review by adiroiban — APPROVED** [review]
Thanks for the refactoring. It looks good.

Only a very minor comment.

Thanks!

File: src/twisted/web/server.py (hunk 1 above)
Comment (itamarst) [reply src/twisted/web/server.py:35]: It's used in a few places still, I didn't remove it from places where it was less common.

File: src/twisted/web/server.py [hunk 2]
```diff
@@ -34,7 +34,14 @@
...
 from twisted.web.error import UnsupportedMethod
-from twisted.web.http import unquote
+from twisted.web.http import (
+    NO_CONTENT,
+    NOT_MODIFIED,
+    HTTPFactory,
+    Request as BaseRequest,
```
Comment (twm) [inline src/twisted/web/server.py:41]: Shouldn't this be `as _BaseRequest` to avoid creating another public name for the type?

Comment (adiroiban) [reply src/twisted/web/server.py:41]: Thanks Tom for the review.

I have created a [separate issue](https://github.com/twisted/twisted/issues/12241) and I will send a PR soon.

**Pull request merged**